modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/mime/JsonUrlEncoder.kt

```diff
@@ -0,0 +1,131 @@
+package com.sphereon.oid.fed.common.mime
+
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonElement
+
+private val qpAllowedChars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_.~".toSet()
+
+/**
+ * URL encode a String.
+ * Converts characters not allowed in URL query parameters to their percent-encoded equivalents.
+ *
+ * @return URL encoded String
+ */
+fun urlEncodeValue(input: String): String {
```

Comment:
Please anotate the function with @jsexport

Reply:
Fixed in next commit